applications/tari_validator_node/src/p2p/services/mempool/service.rs

```diff
@@ -53,43 +60,51 @@ impl MempoolService {
         mempool_requests: mpsc::Receiver<MempoolRequest>,
         outbound: OutboundMessaging,
         tx_valid_transactions: broadcast::Sender<(Transaction, ShardId)>,
+        epoch_manager: EpochManagerHandle,
+        node_identity: Arc<NodeIdentity>,
     ) -> Self {
         Self {
             transactions: Arc::new(Mutex::new(Vec::new())),
             new_transactions,
             mempool_requests,
             outbound,
             tx_valid_transactions,
+            epoch_manager,
+            node_identity,
         }
     }
 
-    pub async fn run(mut self) {
+    pub async fn run(mut self) -> Result<(), MempoolError> {
```

Comment:
Any error will "crash" the mempool service. This run loop should remain active until the new_transactions  and the mempool_requests channels are closed

Reply:
I see, will refactor shortly !